shell/platform/fuchsia/flutter/tests/fakes/scenic/fake_flatland.cc

```diff
@@ -234,6 +267,28 @@ void FakeFlatland::SetOrientation(
   transform->orientation = orientation;
 }
 
+void FakeFlatland::SetOpacity(
+    fuchsia::ui::composition::TransformId transform_id,
+    float value) {
+  if (transform_id.value == 0) {
+    // TODO(fxb/85619): Raise a FlatlandError here
+    FML_CHECK(false) << "FakeFlatland::SetOpacity: TransformId 0 is invalid.";
+    return;
+  }
+
+  auto found_transform = pending_graph_.transform_map.find(transform_id.value);
+  if (found_transform == pending_graph_.transform_map.end()) {
+    // TODO(fxb/85619): Raise a FlatlandError here
+    FML_CHECK(false) << "FakeFlatland::SetOpacity: TransformId "
+                     << transform_id.value << " does not exist.";
+    return;
+  }
+
+  if (value < 0.f || value > 1.f) {
+    FML_CHECK(false) << "FakeFlatland::SetOpacity: Invalid opacity value.";
+  }
```

Comment:
Why doesn't this method set a field on the FakeTransform, and why are there tests for the scale but not tests for the opacity?

Reply:
added in now -- needed to discuss with Emircan on best approach